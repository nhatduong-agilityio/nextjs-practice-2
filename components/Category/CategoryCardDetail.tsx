// Libs
import React, { memo } from 'react'
import NextLink from 'next/link'
import { Button, Heading } from '@chakra-ui/react'

// Components
import ContainerCategoryCard from '@components/Category/ContainerCategoryCard'
import ArrowForwardIcon from '@components/Icons/ArrowForwardIcon'

interface IProps {
  srcImage: string
  imageAlt: string
  name: string
  pathLink: string
}

const CategoryCardDetail = memo(({ srcImage, imageAlt, name, pathLink }: IProps) => (
  <ContainerCategoryCard
    sxContainer={{
      w: 'full',
    }}
    sxWrapperImage={{
      w: 'full',
      height: ['200', '500'],
      overflow: 'hidden',
      borderRadius: '5px',
    }}
    srcImage={srcImage}
    imageAlt={imageAlt}
    sxWrapperChildren={{
      justifyContent: 'flex-start',
      alignItems: 'flex-end',
      backgroundColor: 'rgba(0, 0, 0, 0.4)',
      color: 'white',
      p: ['20px 15px', '50px 40px'],
    }}
  >
    <NextLink href={pathLink} passHref>
      <Heading
        variant={['extraBoldSM', 'extraBoldXL']}
        color='white'
        pb={['10px', '30px']}
        lineHeight='normal'
      >
        {name}
      </Heading>
      <Button
        rightIcon={<ArrowForwardIcon width='14px' height='10px' viewBox='0 0 14 10' fill='white' />}
        variant='ghost'
        color='white'
        size={['xs', 'md']}
        justifyContent='start'
        height={['16px', '27px']}
        style={{
          paddingInline: 0,
        }}
      >
        Shop Now
      </Button>
    </NextLink>
  </ContainerCategoryCard>
))

CategoryCardDetail.displayName = 'CategoryCardDetail'

export default CategoryCardDetail
