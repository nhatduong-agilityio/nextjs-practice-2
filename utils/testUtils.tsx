import { ChakraProvider } from '@chakra-ui/react'
import { AuthProvider } from '@store/providers/AuthProvider'
import { LoadingProvider } from '@store/providers/LoadingProvider'
import type { RenderOptions, RenderResult } from '@testing-library/react'
import { render } from '@testing-library/react'
import { customTheme } from '@themes/index'
import type { ReactElement } from 'react'

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return (
    <ChakraProvider theme={customTheme}>
      <LoadingProvider>
        <AuthProvider>{children}</AuthProvider>
      </LoadingProvider>
    </ChakraProvider>
  )
}

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>): RenderResult =>
  render(ui, { wrapper: AllTheProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
