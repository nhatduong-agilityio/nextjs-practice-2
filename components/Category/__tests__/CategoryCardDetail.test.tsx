// Utils
import { render } from '@utils/testUtils'

import { CategoriesMock } from '@mock/dataMock'

// Components
import CategoryCardDetail from '../CategoryCardDetail'

describe('CategoryCardDetail render', () => {
  const propsMock = {
    srcImage: CategoriesMock.CHAIR.OFFICE.BG_IMAGE,
    imageAlt: CategoriesMock.CHAIR.OFFICE.IMAGE_ALT,
    name: CategoriesMock.CHAIR.OFFICE.NAME,
    pathLink: '/',
  }

  it('Should CategoryCardDetail match Snapshot', () => {
    const { container } = render(<CategoryCardDetail {...propsMock} />)

    expect(container).toMatchSnapshot()
  })
})
