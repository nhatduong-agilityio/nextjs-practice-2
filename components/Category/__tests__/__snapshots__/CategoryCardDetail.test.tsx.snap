// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryCardDetail render Should CategoryCardDetail match Snapshot 1`] = `
<div>
  <div
    class="css-1ki54i"
  >
    <div
      class="css-1ysm64"
    >
      <span
        style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
      >
        <img
          alt="Picture of furniture office chair category"
          data-nimg="fill"
          decoding="async"
          sizes="100vw"
          src="/_next/image?url=%2Fimg.jpg&w=3840&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=640&q=75 640w, /_next/image?url=%2Fimg.jpg&w=750&q=75 750w, /_next/image?url=%2Fimg.jpg&w=828&q=75 828w, /_next/image?url=%2Fimg.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimg.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimg.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimg.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimg.jpg&w=3840&q=75 3840w"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
        />
      </span>
      <div
        class="css-5lowhs"
      >
        <a
          href="/"
        >
          <h2
            class="chakra-heading css-1wzltlp"
          >
            Office Chair
          </h2>
          <button
            class="chakra-button css-154azm5"
            style="padding-inline: 0;"
            type="button"
          >
            Shop Now
            <span
              class="chakra-button__icon css-1hzyiq5"
            >
              <svg
                aria-hidden="true"
                class="chakra-icon css-1cbxkj6"
                focusable="false"
                viewBox="0 0 14 10"
              >
                <path
                  clip-rule="evenodd"
                  d="M12.7264 4.99998C12.7264 4.83123 12.6593 4.66939 12.54 4.55006C12.4207 4.43074 12.2589 4.3637 12.0901 4.3637L0.637071 4.3637C0.468319 4.3637 0.306479 4.43074 0.187153 4.55006C0.0678272 4.66939 0.00079155 4.83123 0.00079155 4.99998C0.00079155 5.16873 0.0678272 5.33057 0.187153 5.4499C0.306479 5.56922 0.468319 5.63626 0.637071 5.63626L12.0901 5.63626C12.2589 5.63626 12.4207 5.56922 12.54 5.4499C12.6593 5.33057 12.7264 5.16873 12.7264 4.99998Z"
                  fill-rule="evenodd"
                />
                <path
                  clip-rule="evenodd"
                  d="M13.8131 5.45047C13.8723 5.39136 13.9193 5.32115 13.9514 5.24385C13.9835 5.16655 14 5.08368 14 4.99998C14 4.91629 13.9835 4.83342 13.9514 4.75612C13.9193 4.67882 13.8723 4.6086 13.8131 4.5495L9.99539 0.73182C9.87592 0.612344 9.71387 0.545223 9.54491 0.545223C9.37594 0.545223 9.2139 0.612344 9.09442 0.73182C8.97494 0.851296 8.90782 1.01334 8.90782 1.18231C8.90782 1.35127 8.97494 1.51332 9.09442 1.63279L12.4629 4.99998L9.09442 8.36717C8.97494 8.48665 8.90782 8.64869 8.90782 8.81766C8.90782 8.98662 8.97494 9.14867 9.09442 9.26814C9.2139 9.38762 9.37594 9.45474 9.54491 9.45474C9.71387 9.45474 9.87592 9.38762 9.99539 9.26814L13.8131 5.45047Z"
                  fill-rule="evenodd"
                />
              </svg>
            </span>
          </button>
        </a>
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
